src/plugins/vis_type_vega/public/data_model/search_api.ts

```diff
@@ -40,9 +40,16 @@ export class SearchAPI {
     return combineLatest(
       searchRequests.map((request) => {
         const requestId = request.name;
-        const params = getSearchParamsFromRequest(request, {
+        const paramsFromRequest = getSearchParamsFromRequest(request, {
           getConfig: this.dependencies.uiSettings.get.bind(this.dependencies.uiSettings),
         });
+        const params = {
+          ...paramsFromRequest,
+          body: {
+            ...paramsFromRequest.body,
+            runtime_mappings: request.runtimeFields,
+          },
+        };
```

Comment:
I think we can just use the following => `params.body.runtime_mappings = request.runtimeFields`. Or do you need to create a new object?

Reply:
Seems there's no need, thanks!